src/storage/mod.rs

```diff
@@ -164,45 +212,76 @@ impl Table {
         Ok(())
     }
 
-    fn open_file(database: &str, table: &str, mode: FileMode) -> Result<File, DatabaseError> {
-        // create new file or open new one
-        let st = format!("{}/{}",database,table);
-        let path = Path::new(&st);
-
-
-        match mode {
-            FileMode::SaveDefault => OpenOptions::new()
-                .write(true)
-                .create(true)
-                .open(path),
-            FileMode::LoadDefault => OpenOptions::new()
-                .read(true)
-                .open(path),
-        }.map_err(|e| e.into())
-    }
-
+    /// Returns columns of table as array
     pub fn columns(&self) -> &[Column] {
         // TODO: Return real columns
         &self.columns
     }
 
-    pub fn add_column(&mut self, name: &str, dtype: DataType) {
+    /// Adds a column to the tabel
+    /// Returns name of Column or on fail DatabaseError
+    pub fn add_column(&mut self, name: &str, dtype: DataType) -> Result<String, DatabaseError> {
+        match self.columns.iter().find(|x| x.name == name) {
+            Some(_x) => {
+                info!(target: "Column add event", "Column already exists");
+                return Err(DatabaseError::AddColumn)
+            },
+            None => {
+                info!(target: "Column add event", "Column was added");
+            },
+        }
         self.columns.push(Column::create_new(name, dtype));
+        Ok(name.into())
```

Comment:
This is somehow not necessary. Return `Result<(), DatabaseError>` as type instead